packages/source-iottwinmaker/src/property-value/provider.ts

```diff
@@ -0,0 +1,76 @@
+import { DataRequest, Provider, ProviderObserver, DataBase, DataStreamBase, Primitive } from '@iot-app-kit/core';
+import { QueryClient, QueryObserver } from '@tanstack/react-query';
+
+import { IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { getPropertyValueByEntity } from './client/getPropertyValueByEntity';
+import { TwinMakerStaticDataQuery } from './types';
+
+/**
+ * Provider for TwinMaker time series data
+ */
+export class TwinMakerPropertyValueDataProvider implements Provider<DataBase[]> {
+  public tmClient: IoTTwinMakerClient;
+  public queryClient: QueryClient;
+
+  public observer: QueryObserver<DataBase[], unknown, DataBase[], DataBase[], TwinMakerStaticDataQuery[]>;
+
+  private _unsubscribes: (() => void)[] = [];
+
+  constructor(
+    queryClient: QueryClient,
```

Comment:
Instead of instantiating the tmClient and queryClient inside the constructor, Consider them passing as dependencies to the class for better flexibility.

Reply:
they are parameters passed in right now